pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientCnx.java

```diff
@@ -1243,6 +1254,13 @@ public void close() {
        }
     }
 
+    public void close(Throwable e) {
```

Comment:
Is there a particular reason to make this public? `close` as a method name is very generic.

How about 
```suggestion
    private void closeWithException(Throwable e) {
```

Reply:
Yes, it should be `protected`.